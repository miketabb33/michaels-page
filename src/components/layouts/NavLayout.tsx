import React from 'react'
import styled from 'styled-components'
import { navConfig } from '../../config/navConfig'
import { Breakpoint, MQ } from '../../Breakpoint'
import { ChildrenProp } from '../../types/ChildrenProp'
import MobileNav from '../mobile-nav/MobileNav'
import Sidebar from '../sidebar/Sidebar'

const Container = styled.div`
  overflow-x: hidden;
  height: 100%;
  display: grid;
  grid-template-columns: auto;
  grid-template-rows: 40px 1fr;

  @media ${MQ(Breakpoint.tablet)} {
    grid-template-columns: 120px 1fr;
    grid-template-rows: auto;
  }
`

const NavLayout = ({ children }: ChildrenProp) => {
  return (
    <Container>
      <MobileNav navConfig={navConfig} />
      <Sidebar navConfig={navConfig} />
      <main>{children}</main>
    </Container>
  )
}

export default NavLayout
