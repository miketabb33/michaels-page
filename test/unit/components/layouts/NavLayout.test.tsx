import React from 'react'

import NavLayout from '../../../../src/components/layouts/NavLayout'
import { shallowRenderer } from '../../../shallowRenderer'

describe('App', () => {
  it('should match snapshot', () => {
    const renderer = shallowRenderer(<NavLayout>AnyLayout</NavLayout>)
    expect(renderer).toMatchSnapshot()
  })
})
