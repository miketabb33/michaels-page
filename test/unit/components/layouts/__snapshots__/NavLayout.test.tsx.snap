// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<styled.div>
  <MobileHeader
    navConfig={
      [
        {
          "label": "Home",
          "linkTo": "/",
        },
        {
          "label": "Books",
          "linkTo": "/recommended-books",
        },
        {
          "label": "Trading",
          "linkTo": "/my-trading-card-worth",
        },
        {
          "label": "Pong",
          "linkTo": "/pong",
        },
        {
          "label": "T T T",
          "linkTo": "/tic-tac-toe",
        },
      ]
    }
  />
  <Sidebar
    navConfig={
      [
        {
          "label": "Home",
          "linkTo": "/",
        },
        {
          "label": "Books",
          "linkTo": "/recommended-books",
        },
        {
          "label": "Trading",
          "linkTo": "/my-trading-card-worth",
        },
        {
          "label": "Pong",
          "linkTo": "/pong",
        },
        {
          "label": "T T T",
          "linkTo": "/tic-tac-toe",
        },
      ]
    }
  />
  <styled.main>
    AnyLayout
  </styled.main>
</styled.div>
`;
